compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -12759,6 +12960,12 @@ TR::Node *ificmpeqSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier
 
 
    addressCompareConversion(node, s);
+
+   if (s->getLastRun() && !s->comp()->getOption(TR_disableIntegerCompareSimplification))
```

Comment:
Also, does this transformation have to be restricted to the last run? Seems like it could happen earlier